content/en/docs/setup/additional-setup/customize-installation-helm/index.md

```diff
@@ -0,0 +1,154 @@
+
+## Advanced Helm Chart Customization
+
+Istio's helm chart tries to incorporate most of the attributes needed by users for their specific requirements. However, it does not
+contain every possible Kubernetes value you may want to tweak. While it is not practical to have such a mechanism in place, in this
+document we will demonstrate a method which would allow you to do some advanced helm chart customization without the need to directly
+modify Istio's helm chart. Using Helm post rendering capability, you can tweak the installation manifests to meet your requirements easily.
+Post rendering gives the flexibility to manipulate, configure, and/or validate rendered manifests before they are installed by Helm.
+This enables users with advanced configuration needs to use tools like Kustomize to apply configuration changes without the need
+for any additional support from the original chart maintainers.
+
+In this example, we will add a `sysctl` value to Istio’s `ingress-gateway` deployment. We are going to:
+
+1. Create a `sysctl` deployment customization patch template.
+1. Apply the patch using Helm post rendering.
+1. Verify that the `sysctl` patch was correctly applied to the pods.
+
+## Create the Kustomization
+
+First, we create a `sysctl` patch file, adding a `securityContext` to the `ingress-gateway` pod with the additional attribute:
+
+{{< text bash >}}
+$ cat > sysctl-ingress-gw-customization.yaml <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ingress-gateway
+spec:
+  template:
+    spec:
+      securityContext:
+          sysctls:
+          - name: net.netfilter.nf_conntrack_tcp_timeout_close_wait
+            value: "10"
+EOF
+{{< /text >}}
+
+The below shell script helps to bridge the gap between Helm post renderer and Kustomize, as the former works with `stdin/stdout`
+and latter works with files.
+
+{{< text bash >}}
+$ cat > kustomize.sh <<EOF
+#!/bin/sh
+cat > base.yaml
+exec kubectl kustomize # you can also use "kustomize build ." if you have it installed.
+EOF
+chmod +x ./kustomize.sh
+{{< /text >}}
+
+Finally, let us create the `kustomization` yaml file, which is the input for `kustomize`
+with the set of resources and associated customization details.
+
+{{< text bash >}}
+$ cat > kustomization.yaml <<EOF
+resources:
+- base.yaml
+patchesStrategicMerge:
+- sysctl-ingress-gw-customization.yaml
+EOF
+{{< /text >}}
+
+## Apply the Kustomization
+
+Now that the Kustomization file is ready, let us use Helm to make sure this gets applied properly.
+
+### Add the Helm repository for Istio
+
+{{< text bash >}}
+$ helm repo add istio https://istio-release.storage.googleapis.com/charts
+$ helm repo update
+{{< /text >}}
+
+### Render and Verify using Helm Template
+
+We can use Helm `post-renderer` to validate rendered manifests before they are installed by Helm
+
+{{< text bash >}}
+$ helm template istio/gateway --post-renderer ./kustomize.sh | grep -B 4 -A 1 netfilter.nf_conntrack_tcp_timeout_close_wait
```

Comment:
I'm getting this when I run this command

```bash

Error: error while running post render on files: error while running command /home/jacob/cluster/kind/kustomize.sh. error output:
error: no matches for Id Deployment.v1.apps/ingress-gateway.[noNs]; failed to find unique target for patch Deployment.v1.apps/ingress-gateway.[noNs]
: exit status 1
```

I think you want the following as you aren't using `nf_conntrack_tcp_timeout_close_wait`

```suggestion
$ helm template istio/gateway --post-renderer ./kustomize.sh
```

Reply:
I'm still getting the same error when I looked at the updated page.

Reply:
@jacob-delgado have you verified if this change now works for you?

Reply:
It still doesn't work for me. I'm copying the commands from the webpage. 

```shell
jacob@jacob-build-3:~/faseela$ cat > sysctl-ingress-gw-customization.yaml <<EOF
> apiVersion: apps/v1
> kind: Deployment
> metadata:
>   name: istio-ingressgateway
> spec:
>   template:
>     spec:
>       securityContext:
>           sysctls:
>           - name: net.netfilter.nf_conntrack_tcp_timeout_close_wait
>             value: "10"
> EOF
jacob@jacob-build-3:~/faseela$
jacob@jacob-build-3:~/faseela$ cat > kustomize.sh <<EOF
> #!/bin/sh
> cat > base.yaml
> exec kubectl kustomize # you can also use "kustomize build ." if you have it installed.
> EOF
jacob@jacob-build-3:~/faseela$ chmod +x ./kustomize.sh
jacob@jacob-build-3:~/faseela$ cat > kustomization.yaml <<EOF
> resources:
> - base.yaml
> patchesStrategicMerge:
> - sysctl-ingress-gw-customization.yaml
> EOF
jacob@jacob-build-3:~/faseela$ helm repo add istio https://istio-release.storage.googleapis.com/charts
"istio" already exists with the same configuration, skipping
jacob@jacob-build-3:~/faseela$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "istio" chart repository
...Successfully got an update from the "jetstack" chart repository
...Successfully got an update from the "gloo" chart repository
Update Complete. ⎈Happy Helming!⎈
jacob@jacob-build-3:~/faseela$ helm template istio/gateway --post-renderer ./kustomize.sh | grep -B 4 -A 1 netfilter.nf_conntrack_tcp_timeout_close_wait
Error: error while running post render on files: error while running command /home/jacob/faseela/kustomize.sh. error output:
error: no matches for Id Deployment.v1.apps/istio-ingressgateway.[noNs]; failed to find unique target for patch Deployment.v1.apps/istio-ingressgateway.[noNs]
: exit status 1

Use --debug flag to render out invalid YAML
```

Reply:
When I do the same with the solo's guide at https://docs.solo.io/gloo-edge/latest/installation/gateway/kubernetes/helm_advanced/ it works just fine for me.

I am copying and pasting it with the webpage. I'm not typing this out manually. Maybe a copy/paste issue?

Reply:
Thanks @jacob-delgado 

I just tried the same here and also see the error on this page. I did not try the solo.io page (Update: I did try the solo.io page and it worked).

Reply:
I will get back to this soon..on vacation currently.

Reply:
OK. Comparing things, nothing jumps out as a cut/paste error. Maybe something related to the helm chart/repo.

Reply:
Ok, I figured out the problem

```shell
$ helm template istio/gateway --post-renderer ./kustomize.sh
```

should be

```shell
$ helm template istio-ingressgateway istio/gateway --post-renderer ./kustomize.sh
```

Reply:
> Ok, I figured out the problem cc @kfaseela 
> 
> ```shell
> $ helm template istio/gateway --post-renderer ./kustomize.sh
> ```
> 
> should be
> 
> ```shell
> $ helm template istio-ingressgateway istio/gateway --post-renderer ./kustomize.sh
> ```

Thanks Jacob! I will try this out next week once am back from vacation and update the PR!